### PR TITLE
Adds a Dockerfile for local development

### PR DESCRIPTION
**Pull request by danielhartnell**
```
λ ~/ docker build -t linkhack .
λ ~/ docker run -p 3000:3000 linkhack
=> Booting Puma
=> Rails 6.1.7.7 application starting in development
=> Run `bin/rails server --help` for more startup options
Puma starting in single mode...
* Puma version: 6.1.0 (ruby 3.2.1-p31) ("The Way Up")
*  Min threads: 5
*  Max threads: 5
*  Environment: development
*          PID: 1
* Listening on http://0.0.0.0:3000
Use Ctrl-C to stop
```

It already uses SQLite so no need to introduce docker compose to bring up supporting resources like Postgres.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM ruby:3.2.1-bullseye
```
Comment (khaleksa) [inline Dockerfile:1]: Please update to ruby version 3.2.4

**Review by khaleksa — APPROVED** [review]
LGTM! Thank you for updating ruby version

**Comment by iMartzen** [comment]
Why is this not merged?

**Comment by danielhartnell** [comment]
I don't have access to merge so it's up to someone with write permission @iMartzen

**Comment by martijnrusschen** [comment]
Done

**Pull request merged**